scrapy_poet/middleware.py

```diff
@@ -35,8 +37,20 @@ def process_request(self, request: Request, spider):
         return utils.DummyResponse(url=request.url, request=request)
 
     @inlineCallbacks
-    def process_response(self, request: Request, response, spider):
-        # find out the dependencies
+    def process_response(self, request: Request, response: Response,
+                         spider: Spider):
+        """This method instantiates all ``Injectable`` sub-classes declared as
+        request callback arguments and any other parameter with a provider for
+        its type. Otherwise, this middleware doesn't populate
+        ``request.cb_kwargs`` for this argument.
+
+        .. warning::
+
+            Currently, we're able to inject any type that inherits from
+            Injectable, but the only external dependency that we're able to
+            build and inject is the ``scrapy.Response``.
```

Comment:
`ResponseData` can also be injected even if it is not an Injectable. Also, types with a defined provider would be injected.

Reply:
Fixed. I've just updated this docstring to clarify what should be said. 

```rst
We should be able to inject any type into classes that inherit from
``web_poet.pages.Injectable``, but currently, we're only able to
build and inject ``scrapy.Response`` instances.
```